connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedConfig.java

```diff
@@ -257,7 +257,6 @@
                     CommonClientConfigs.DEFAULT_SECURITY_PROTOCOL,
                     ConfigDef.Importance.MEDIUM,
                     CommonClientConfigs.SECURITY_PROTOCOL_DOC)
-            .withClientSslSupport()
             .withClientSaslSupport()
```

Comment:
Is SASL support significantly different and only needed in distributed mode?

Reply:
Good question. SASL support is different in that it's only useful for connecting to Kafka, as opposed to SSL, which can also be used for performing authentication at the REST API level. But both types of worker should support all SASL-related configs since they both do connect to Kafka.

However, as far as I can tell, the lack of definition for SASL-related configs in the `StandaloneConfig` class isn't causing any bugs at the moment, so in order to be conservative with the changes here I opted to just move the SSL-related configs into the parent config class.

I think there's a lot of configs that are defined in the `DistributedConfig` class that would make more sense in the `WorkerConfig` class, and possibly some configs in the `WorkerConfig` class that are really only relevant for distributed mode (`rest.advertised.*` seems strange, given that standalone workers don't have any other workers to advertise their URLs to...). We can address all of that here if there are other issues caused by them, but if it's more of a stylistic thing and it's alright with you, I think a follow-up ticket for cleaning up where we define these configs might be a better way to handle the non-urgent work and get in this somewhat-more-urgent fix now. WDYT?

Reply:
I think that makes sense.